framework/src/media/MediaPlayerImpl.cpp

```diff
@@ -291,31 +291,38 @@ player_result_t MediaPlayerImpl::stop()
 		return PLAYER_ERROR;
 	}
 
-	mpw.enQueue(&MediaPlayerImpl::stopPlayer, shared_from_this());
+	mpw.enQueue(&MediaPlayerImpl::stopPlayer, shared_from_this(), 0);
 
 	return PLAYER_OK;
 }
 
-void MediaPlayerImpl::stopPlayer()
+void MediaPlayerImpl::stopPlayer(int errcode)
```

Comment:
1. instead of int errcode, How about recorder_result_t result & player_result_t result
2. then, errcode == 0 -> result != PLAYER_OK
3. add todo in eum of result_e